### ISSUE TITLE
Verify = false option is not being respected

### ISSUE THREAD

**Issue by ftdysa** [guzzle/guzzle#1154]
It doesn't look like the following works anymore:

```php
$client = new Client(['verify' => false]);
```

I know all about the security implications but I've got to make this work for a short period of time.

Here is a sample script that shows it being broken ...

```php
<?php

require 'vendor/autoload.php';

$ch = curl_init('https://example.com');
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);

$result = curl_exec($ch); // This works

$client = new \GuzzleHttp\Client(['verify' => false]);
$response = $client->get('https://example.com'); // This does not work
````

Also, CurlFactory has a warning.

```
Warning: curl_setopt_array(): Array keys must be CURLOPT constants or equivalent integer values in /path/to/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php on line 56
```

**Comment by mtdowling** [comment]
Weird. This works on my system without issue. I also don't get any warnings about CURLOPT constants.

Here's my user-agent: `GuzzleHttp/6.0.1 curl/7.42.1 PHP/5.6.9`

```php
$ php -v
PHP 5.6.9 (cli) (built: May 20 2015 17:26:38) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2015, by Zend Technologies
```

Can you set the `debug` option to true and provide the output?

**Comment by ftdysa** [comment]
I'm still seeing an exception (I think its on connect, not request). Here's the full exception trace:

```
fred@home:~/ournatnet$ php test.php      

Warning: curl_setopt_array(): Array keys must be CURLOPT constants or equivalent integer values in /path/to/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php on line 56

Fatal error: Uncaught exception 'GuzzleHttp\Exception\ConnectException' with message 'cURL error 35: error:0D0C50A1:asn1 encoding routines:ASN1_item_verify:unknown message digest algorithm (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)' in /path/to/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:161
Stack trace:
#0 /path/to/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(125): GuzzleHttp\Handler\CurlFactory::createRejection(Object(GuzzleHttp\Handler\EasyHandle), Array)
#1 /path/to/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(90): GuzzleHttp\Handler\CurlFactory::finishError(Object(GuzzleHttp\Handler\CurlMultiHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory))
#2 /path/to/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(180): GuzzleHttp\Handler\CurlFactory::finish(Object(GuzzleHttp\Handler\CurlMultiHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlF in /path/to/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php on line 161
```

I will note that you probably will not encounter this unless you are running an older version of SSL. Turning off `CURLOPT_SSL_VERIFYPEER` will temporarily fix this until I can get the server's openssl upgraded (which is in the works, but in the mean time I need to get this up).

**Comment by mtdowling** [comment]
What version of Guzzle are you using? What version of PHP? What version of cURL?

You may have to play around and step through debugging because I can't reproduce this. On my system, this block of code is being execute as intended (disabling CURLOPT_SSL_VERIFYPEER) https://github.com/guzzle/guzzle/blob/master/src/Handler/CurlFactory.php#L291

**Comment by ftdysa** [comment]
I'll see if I can't step through the MultiHandler and see why this is failing. I know that the `applyHandlerOptions` is being invoked properly so I can only deduce that it's being dropped somewhere along the way.

Versions:
Guzzle: 6.0.1
PHP: 5.5.9
Curl: 7.14.0

**Comment by ftdysa** [comment]
Ahh, I figured it out.

The option `CURLOPT_PROTOCOLS` is only available on curl > 7.19.4.

Removing the line in `CurlFactory::getDefaultConf` that sets makes this work as expected as the warning about curl_setopt_array go away.

**Comment by mtdowling** [comment]
Ok thanks. I'm waiting on a travis build to pass that addresses this very issue. It will be merged in and tagged in the next release (probably this weekend): https://travis-ci.org/guzzle/guzzle/builds/69003504
